### PR TITLE
fix docupdate conflict

### PR DESCRIPTION
**Pull request by sravfeyn**
https://sentry.io/dimagi/commcarehq/issues/242460043/

This issue started occurring after app status report was [recently updated](https://github.com/dimagi/commcare-hq/pull/16219/files#diff-fa0e8af5f7775e2be761383c7e5b33c3R85). I am not sure if there is a better solution than the fix I made
 
FYI @benrudolph @calellowitz

### CODE REVIEW

**Comment by sravfeyn** [comment]
Wait for the build https://travis-ci.org/dimagi/commcare-hq/builds/248358886

**Comment by benrudolph** [comment]
do you think we could use either locking (something like `CriticalSection`) or attempt to save the doc and catch the document update error and when that happens refetch the doc and try to save again?

**Comment by sravfeyn** [comment]
That's much better, I have updated it

File: corehq/apps/ota/utils.py [hunk 1]
```diff
@@ -62,9 +63,15 @@ def reset_demo_user_restore(commcare_user, domain):
...
-    commcare_user.demo_restore_id = demo_restore.id
-    commcare_user.save()
+    # Set reference to new restore
+    try:
+        commcare_user.demo_restore_id = demo_restore.id
+        commcare_user.save()
+    except ResourceConflict:
+        # If CommCareUser.report_metadata gets updated by sync log pillow, after a restore is
+        #   generated or for any other reason, there will be a DocumentUpdate conflict
+        commcare_user = CommCareUser.get(commcare_user.get_id)
```
Comment (benrudolph) [inline corehq/apps/ota/utils.py]: sweet, i think you'll still need to set the `demo_restore_id`

Comment (sravfeyn) [inline corehq/apps/ota/utils.py]: Sorry, thanks.

**Review by benrudolph — APPROVED** [review]

**Pull request merged**